### PR TITLE
chore: Extract SwitchChangeEventHandler and SwitchSize

### PR DESCRIPTION
**Pull request by yesmeck**
With this change, we can define a change handler more convenience:

Before:

```typescript
const handleChange = (checked: boolean, event: MouseEvent) => {
  ...
}
```

After:

```typescript
const handleChange: SwitchChangeEventHandler = (checked, event) => {
  ...
}
```

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [x] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |Improve type definitions of Switch|
| 🇨🇳 Chinese |改进 Switch 的类型定义|

### ☑️ Self Check before Merge

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

### CODE REVIEW

File: components/switch/index.tsx [hunk 1]
```diff
@@ -7,14 +7,18 @@ import Wave from '../_util/wave';
 import Icon from '../icon';
 import { ConfigConsumer, ConfigConsumerProps } from '../config-provider';
 
+export type SwitchSize = 'small' | 'default';
+export type SwitchChangeEventHandler = (checked: boolean, event: MouseEvent) => void;
+export type SwitchClickEventHandler = SwitchChangeEventHandler;
+
 export interface SwitchProps {
   prefixCls?: string;
-  size?: 'small' | 'default';
```
Comment (afc163) [inline components/switch/index.tsx]: 这种有没有更好的声明方式，Enum 能用不？

Comment (yesmeck) [reply components/switch/index.tsx]:
```typescript
export enum SwitchSize {
  SMALL = 'small';
  DEFAUL = 'default';
}
```

```typescript
import { SwitchSize } from 'antd/es/switch';

<Switch size={SwitchSize.SMALL} />
```

Comment (afc163) [reply components/switch/index.tsx]: 这样就改使用方式了。

目前看来最好的方式是：https://github.com/ant-design/ant-design/issues/7965#issuecomment-506191378

Comment (ztplz) [reply components/switch/index.tsx]: 他那种assert不推荐的，毕竟assert能少用就少用。如果值A是B类型，你应该直接声明A是B类型，而不是把A里的类型都assert成B里的属性然后通过ts的类型检查。

**Pull request merged**